cli/command/cli.go

```diff
@@ -407,9 +418,55 @@ type ClientInfo struct {
 	DefaultVersion  string
 }
 
-// NewDockerCli returns a DockerCli instance with IO output and error streams set by in, out and err.
-func NewDockerCli(in io.ReadCloser, out, err io.Writer, isTrusted bool, containerizedFn func(string) (clitypes.ContainerizedClient, error)) *DockerCli {
-	return &DockerCli{in: NewInStream(in), out: NewOutStream(out), err: err, contentTrust: isTrusted, newContainerizeClient: containerizedFn}
+// NewDockerCli returns a DockerCli instance with all operators applied on it.
+func NewDockerCli(ops ...DockerCliOperator) *DockerCli {
+	cli := &DockerCli{}
```

Comment:
Does this mean that some `ops` are always needed (e.g. to provide a stream). Would we want to stub those out to some default (either `os.Std*` or `io.Discard` I suppose)?

Reply:
I added default operations:
```go
func NewDockerCli(ops ...DockerCliOperator) (*DockerCli, error) {
	cli := &DockerCli{}
	defaultOps := []DockerCliOperator{
		WithStandardStreams(),
		WithContentTrustFromEnv(),
		WithContainerizedClient(containerizedengine.NewClient),
	}
	ops = append(defaultOps, ops...)
	if err := cli.Apply(ops...); err != nil {
		return nil, err
	}
	return cli, nil
}
```

Reply:
One can override them as users operations are applied after the default ones.

Reply:
I think the current approach looks valid; there may other caveats (what if want _without_ containerised client, or without content-trust)"

possible alternatives are;

- only apply defaults if `ops` is empty
- add a `WithDefaults()` option (that's applying these), which can be used if someone wants the defaults _plus_ some custom configurations.

But we can discuss those solutions (not sure which approach makes the most sense)